experiment/src/org/labkey/experiment/ExperimentUpgradeCode.java

```diff
@@ -222,16 +224,30 @@ public static void generateExpProtocolApplicationEntityIds(ModuleContext ctx) th
         {
             var protocolApplicationTable = ExperimentService.get().getTinfoProtocolApplication();
             List<ProtocolApplication> applications = new TableSelector(protocolApplicationTable).getArrayList(ProtocolApplication.class);
+            Parameter rowId = new Parameter("rowid", JdbcType.INTEGER);
+            Parameter entityId = new Parameter("entityid", JdbcType.GUID);
+            ParameterMapStatement pm = new ParameterMapStatement(protocolApplicationTable.getSchema().getScope(), tx.getConnection(),
+                    new SQLFragment("UPDATE " + protocolApplicationTable.getSelectName() + " SET EntityId = ? WHERE RowId = ?", entityId, rowId), null);
+            int count = 0;
 
             for (var application: applications)
             {
                 if (application.getEntityId() == null)
                 {
-                    application.setEntityId(new GUID());
-                    Table.update(ctx.getUpgradeUser(), protocolApplicationTable, application, application.getRowId());
+                    rowId.setValue(application.getRowId());
+                    entityId.setValue(new GUID());
+                    pm.addBatch();
+                    count = count + 1;
+
+                    if (count == 1000)
+                    {
+                        count = 0;
+                        pm.executeBatch();
+                    }
                 }
             }
 
+            pm.executeBatch();
```

Comment:
```suggestion
            if (count > 0) pm.executeBatch();
```
I'm not sure whether calling executeBatch() with no rows is an error.